src/cryptonote_protocol/cryptonote_protocol_handler.h

```diff
@@ -132,27 +132,34 @@ namespace cryptonote
 
     //----------------- i_bc_protocol_layout ---------------------------------------
     template<class T>
-    bool relay_on_public_network_generic(typename T::request& arg, cryptonote_connection_context& exclude_context)
+    bool relay_to_synchronized_peers(typename T::request& arg, cryptonote_connection_context& exclude_context)
     {
       LOG_PRINT_L2("[" << epee::net_utils::print_connection_context_short(exclude_context) << "] post relay " << typeid(T).name() << " -->");
-      std::string arg_buff;
-      epee::serialization::store_t_to_binary(arg, arg_buff);
-
       std::vector<std::pair<epee::net_utils::zone, boost::uuids::uuid>> connections;
       m_p2p->for_each_connection([&exclude_context, &connections](connection_context& context, nodetool::peerid_type peer_id, uint32_t support_flags)
       {
-        epee::net_utils::zone zone = context.m_remote_address.get_zone();
-        if (peer_id && exclude_context.m_connection_id != context.m_connection_id && zone == epee::net_utils::zone::public_)
-          connections.push_back({zone, context.m_connection_id});
+        if (context.m_state != cryptonote_connection_context::state_synchronizing)
+        {
+          epee::net_utils::zone zone = context.m_remote_address.get_zone();
+          if (peer_id && exclude_context.m_connection_id != context.m_connection_id && zone == epee::net_utils::zone::public_)
```

Comment:
Why is this only relaying to public zone connections?  If a connection is via tor, for example, shouldn't we still relay it?

Reply:
Currently TOR/I2P mode is only for relaying transactions and they have some scheme to try hide the transmissions of transactions with timing/padding, so I didn't want to introduce relaying of votes/uptimes to this prematurely without somehow compromising what they've set up.